threat_intel/virustotal.py

```diff
@@ -45,6 +45,63 @@ def get_file_reports(self, resources):
         return all_responses
 
     @MultiRequest.error_handling
+    def get_file_behaviour(self, resources):
+        """Retrieves a report about the behaviour of a  md5, sha1, and/or sha2 hash of
+            a file when executed in a sandboxed environment (Cuckoo sandbox).
+
+        Args:
+            resources: list of string hashes.
+        Returns:
+            A dict with the hash as key and the VT report as value.
+        """
+        api_name = 'virustotal-file-behaviour'
+        all_responses, resources = self._bulk_cache_lookup(api_name, resources)
+
+        resource_chunks = self._prepare_resource_chunks(resources)
+        response_chunks = self._request_reports("resource", resource_chunks, 'file/behaviour')
+
+        self._extract_response_chunks(all_responses, response_chunks, api_name)
+
+        return all_responses
+
+    @MultiRequest.error_handling
+    def get_file_download(self, resources):
+        """Retrieves a file from its a md5, sha1, and/or sha2 hash.
+        Args:
+            resources: list of string hashes.
+        Returns:
+            a file download
+        """
+        api_name = 'virustotal-file-download'
+        all_responses, resources = self._bulk_cache_lookup(api_name, resources)
+
+        resource_chunks = self._prepare_resource_chunks(resources)
+        response_chunks = self._request_reports("resource", resource_chunks, 'file/download')
+
+        self._extract_response_chunks(all_responses, response_chunks, api_name)
+
+        return all_responses
+
+    @MultiRequest.error_handling
+    def get_file_network_traffic(self, resources):
```

Comment:
All three methods (`get_file_behaviour`, `get_file_download` and `get_file_network`) follow the same pattern.
I wonder if that could be extracted into a method to avoid the duplication?